src/net_processing.cpp

```diff
@@ -1533,6 +1536,12 @@ void static ProcessGetData(CNode* pfrom, const CChainParams& chainparams, CConnm
             if (mi != mapRelay.end()) {
                 connman->PushMessage(pfrom, msgMaker.Make(nSendFlags, NetMsgType::TX, *mi->second));
                 push = true;
+
+                auto num = mempool.setUnbroadcastTxIDs.erase(inv.hash);
+                if (num) {
+                    LogPrint(BCLog::NET, "Removed %i from setUnbroadcastTxIDs \n", inv.hash.GetHex());
```

Comment:
I think this deserves a comment. We remove the txhash from the set here because a peer has requested it.

Also note that when we rebroadcast the tx INV, that pushes it back into `mapRelay`, so we should be able to relay it and remove it from `setUnbroadcastTxIDs` when it's requested.

Reply:
added comment "Once the first peer requests GETDATA for a txn, we deem initial broadcast a success"

are you suggesting I also document the mapRelay interactions?